pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -471,3 +489,41 @@ func (s *Server) getRootCertFromSecret(name, namespace string) (*istioCredential
 	}
 	return kube.ExtractRoot(secret.Data)
 }
+
+// checkAndRunNonRevisionLeaderElectionIfRequired checks the ConfigMap for leader election and runs non-revision leader election
+// to figure out when the older leader has released the lock.
+// Once the lock is acquired, it will close the activateCh channel to signal the per-revision leader to start writing status.
+func (s *Server) checkAndRunNonRevisionLeaderElectionIfRequired(args *PilotArgs, activateCh chan struct{}) {
```

Comment:
I think this looks good.  Can we add integ tests to demonstrate 1) the new per-revision status population for gateways, and 2) the backwards compatibility which prevents multiple leaders?

Reply:
sure, will add.